src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/UDPCommunicator.java

```diff
@@ -0,0 +1,493 @@
+package com.avispl.symphony.dal.communicator.aver.ptz;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketTimeoutException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import com.avispl.symphony.api.dal.dto.control.ConnectionState;
+import com.avispl.symphony.api.dal.error.CommandFailureException;
+import com.avispl.symphony.dal.BaseDevice;
+import com.avispl.symphony.dal.communicator.Communicator;
+import com.avispl.symphony.dal.communicator.ConnectionStatus;
+
+public class UDPCommunicator extends BaseDevice implements Communicator {
+	protected static final char[] hexArray = "0123456789ABCDEF".toCharArray();
+
+	private DatagramSocket datagramSocket;
+	private int port;
+	private InetAddress address;
+	private int timeout = 4000;
+	private int bufferLength = 24;
+	private List<String> commandErrorList;
+	private List<String> commandSuccessList;
+
+	private final ReadWriteLock lock = new ReentrantReadWriteLock();
+	private final ConnectionStatus status = new ConnectionStatus();
+
+	/**
+	 * Empty constructor
+	 */
+	public UDPCommunicator() {
+		// Do nothing
+	}
+
+	/**
+	 * This method returns the device UPD timeout
+	 *
+	 * @return int This returns the current UDP timeout.
+	 */
+	public int getTimeout() {
+		return this.timeout;
+	}
+
+	/**
+	 * This method is used set the device UDP timeout
+	 *
+	 * @param timeout This is the UDP timeout to set
+	 */
+	public void setTimeout(int timeout) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException("Cannot change properties after init() was called");
```

Comment:
It is a kind of generic error message in this class. Can we move it to the variable?

Reply:
Yes, I think it's possible